### PR TITLE
Feature/2 webpack

### PR DESCRIPTION
**Pull request by timmi05**
Task 2 (Webpack lecture)

### CODE REVIEW

File: common.webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+module.exports = {
+
+    context: path.join(__dirname, 'src'),
+
+    entry: {
+        app: './index.jsx'
+    },
+
+    output: {
+        path: __dirname + '/dist',
```
Comment (ant-plotnikov) [inline common.webpack.config.js]: please use `path` module

File: common.webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+const path = require('path');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const { CleanWebpackPlugin } = require('clean-webpack-plugin');
+
+module.exports = {
+
+    context: path.join(__dirname, 'src'),
+
+    entry: {
+        app: './index.jsx'
```
Comment (ant-plotnikov) [inline common.webpack.config.js]: please use `path.resolve` here not to rely on `/` separator

File: common.webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    module: {
+        rules: [
+            {
+                test: /\.jsx?$/,
+                loader: 'babel-loader',
+                exclude: /node_modules/
+            },
+            {
+                test: /\.css$/,
+                use: [ 'style-loader', 'css-loader'],
```
Comment (ant-plotnikov) [inline common.webpack.config.js]: Please `exclude: /node_modules/` here as well

File: common.webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        app: './index.jsx'
+    },
+
+    output: {
+        path: __dirname + '/dist',
+        filename: 'main.js'
+    },
+
+    resolve: {
+        extensions: ['.js', '.jsx']
```
Comment (ant-plotnikov) [inline common.webpack.config.js]: And could you move this `resolve: {extensions: ['js', 'jsx']}` to `rules` for babel-loader?

File: prod.webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+const TerserJSPlugin = require('terser-webpack-plugin');
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+const OptimizeCSSAssetsPlugin = require('optimize-css-assets-webpack-plugin');
+module.exports = {
+  optimization: {
+    minimizer: [new TerserJSPlugin({}), new OptimizeCSSAssetsPlugin({})],
```
[the remark this answers is not in the document]
Comment (timmi05) [reply prod.webpack.config.js]: Done

File: common.webpack.config.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (timmi05) [reply common.webpack.config.js]: Done

File: common.webpack.config.js (hunk 1 above)
Comment (timmi05) [reply common.webpack.config.js]: Done

File: common.webpack.config.js (hunk 2 above)
Comment (timmi05) [reply common.webpack.config.js]: Done

File: common.webpack.config.js (hunk 3 above)
Comment (timmi05) [reply common.webpack.config.js]: Done

File: common.webpack.config.js (hunk 4 above)
Comment (timmi05) [reply common.webpack.config.js]: Done

**Pull request merged**